packages/upload-client/src/blob.js

```diff
@@ -303,7 +304,46 @@ export async function add(
     })
   }
 
-  return multihash
+  // Ensure the blob has been accepted
+  const acceptReceipt = await new ReceiptPoller(options).poll(
+    nextTasks.accept.task.link()
+  )
+
+  // @ts-ignore Property
+  /* c8 ignore next 5 */
+  if (!acceptReceipt?.out.ok?.site) {
+    throw new Error(`failed ${BlobCapabilities.add.can} invocation`, {
+      cause: 'failed to get blob/accept receipt',
+    })
+  }
+
+  const blocks = new Map(
+    [...acceptReceipt.iterateIPLDBlocks()].map((block) => [
+      `${block.cid}`,
+      block,
+    ])
+  )
+  const site = Delegation.view(
+    {
+      root: /** @type {import('@ucanto/interface').UCANLink} */ (
+        // @ts-ignore Property
+        acceptReceipt.out.ok.site
+      ),
+      blocks,
+    },
+    null
+  )
+  /* c8 ignore next 5 */
+  if (!site) {
+    throw new Error(`failed ${BlobCapabilities.add.can} invocation`, {
+      cause: 'failed to get blob/accept receipt',
+    })
+  }
+
+  return {
+    multihash,
+    site: site.link(),
```

Comment:
we should return the delegation, so that the client can invoke it if intended

Reply:
good point! addressed